### ISSUE TITLE
gsub!': asciidoctor: FAILED to load AsciiDoc document - can't modify frozen String (FrozenError)

### ISSUE THREAD

**Issue by siddjain** [asciidoctor/asciidoctor-mathematical#56]
I am getting this error while trying to use asciidoctor-mathematical from the docker image https://github.com/asciidoctor/docker-asciidoctor. wondering if anyone can help with this.

```
bash-4.4# asciidoctor-pdf -r asciidoctor-mathematical -o test.pdf chap01.adoc --trace
Traceback (most recent call last):
	20: from /usr/bin/asciidoctor-pdf:23:in `<main>'
	19: from /usr/bin/asciidoctor-pdf:23:in `load'
	18: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-pdf-1.5.0.alpha.18/bin/asciidoctor-pdf:31:in `<top (required)>'
	17: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/cli/invoker.rb:111:in `invoke!'
	16: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/cli/invoker.rb:111:in `each'
	15: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/cli/invoker.rb:128:in `block in invoke!'
	14: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/convert.rb:183:in `convert_file'
	13: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/convert.rb:183:in `open'
	12: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/convert.rb:183:in `block in convert_file'
	11: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/convert.rb:78:in `convert'
	10: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/load.rb:83:in `load'
	 9: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/document.rb:555:in `parse'
	 8: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/document.rb:555:in `each'
	 7: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/document.rb:556:in `block in parse'
	 6: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/document.rb:556:in `[]'
	 5: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-mathematical-0.3.0/lib/asciidoctor-mathematical/extension.rb:40:in `process'
	 4: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-mathematical-0.3.0/lib/asciidoctor-mathematical/extension.rb:40:in `each'
	 3: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-mathematical-0.3.0/lib/asciidoctor-mathematical/extension.rb:41:in `block in process'
	 2: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-mathematical-0.3.0/lib/asciidoctor-mathematical/extension.rb:100:in `handle_prose_block'
	 1: from /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-mathematical-0.3.0/lib/asciidoctor-mathematical/extension.rb:135:in `handle_inline_stem'
/usr/lib/ruby/gems/2.5.0/gems/asciidoctor-mathematical-0.3.0/lib/asciidoctor-mathematical/extension.rb:135:in `gsub!': asciidoctor: FAILED: /documents/chap01/chap01.adoc: Failed to load AsciiDoc document - can't modify frozen String (FrozenError)
```

**Comment by siddjain** [comment]
Is it okay to cross-list this issue at https://github.com/asciidoctor/docker-asciidoctor? I am not getting any response here

**Comment by ProgramFan** [comment]
asciidoctor-mathematical shall be updated not to modify string in-place. Have you tested the version on master?

**Comment by siddjain** [comment]
we are using the docker image. could someone please update the docker image with the fix? is there link to a bug and changeset that fixes this problem?

________________________________

**Comment by ProgramFan** [comment]
It seems the issue is fixed in master and we need a new release. Would you please make a release from the current master to fix the frozen-string issue? @tstumm

**Comment by ProgramFan** [comment]
Meanwhile, you can update the docker file and build the docker image your self. Even after we make the release, you still have to do it to get a new docker image, or you have to wait until they make a release. @siddjain

**Comment by siddjain** [comment]
what are the steps for me to build a new docker image myself (which has this fix) so that I am not blocked by a new release? thanks

________________________________

**Comment by ProgramFan** [comment]
You will have to learn a bit docker and Google can help youl

**Comment by siddjain** [comment]
could the changeset that fixes this bug be please linked to this thread?

________________________________

**Comment by siddjain** [comment]
ping on this. any ETA when a new version will be released?

________________________________

**Comment by tstumm** [comment]
Try updating to 0.3.1.

**Comment by siddjain** [comment]
I get this with 0.3.1

```
bash-4.4# asciidoctor-pdf -r asciidoctor-mathematical -o test.pdf chap01.adoc --trace
asciidoctor: WARNING: conversion missing in backend pdf for stem
asciidoctor: WARNING: conversion missing in backend pdf for stem
asciidoctor: WARNING: conversion missing in backend pdf for stem
```

and the math does not render in the pdf

**Comment by siddjain** [comment]
docker build log can be seen here https://gist.github.com/siddjain/e623c801157a12a37c04b8d84c070d9b
it says
Successfully installed asciidoctor-mathematical-0.3.1

**Comment by siddjain** [comment]
also this:

bash-4.4# ls /usr/lib/ruby/gems/2.5.0/gems/asciidoctor-mathematical-0.3.1/lib
asciidoctor-mathematical     asciidoctor-mathematical.rb

________________________________

**Comment by siddjain** [comment]
ping.. any update on this?

________________________________

**Comment by siddjain** [comment]
is there any update on this? its been a while since this issue was raised.

**Comment by mojavelinux** [comment]
I have included a fix for this in #92.
